students/etwum/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,153 @@
+            str_donation_choice = input("Would you like to add another donation? (yes or no) ")
+            print('-----------------------------------------------------')
+            print()
+            if str_donation_choice == 'yes':
+                donate_more = float(input("How much would you like to donate? "))
+                total_donations = x["total donations"] + donate_more
+                count_donations = x["number donations"] + 1
+                avg_donation = total_donations/count_donations
+                x["total donations"] = total_donations
+                x["number donations"] = count_donations
+                x["avg donation"] = '{:.2f}'.format(avg_donation)
+
+            email = "Dear {a},\n\nThank you for your generous donations of ${b:.2f} to our charity.\n".format(
+                a=x["donor name"],b=x["total donations"])
+            print(email)
+            break
+
+    # if the donor's name is not in the list it adds the name to the list and ask for a donation amount
+    # then prints an email to the new donor
+    else:
+
+        print("Donor not found")
+        new_donation_amount = None
+        boolValid = False
+        while boolValid != True:
+            try:
+                new_donation_amount = float(input("Please input a donation "
+                                                  "amount in order to add the donor to the list. "))
+            except ValueError as e:
+                print(e)
+                print("Please input a valid number")
+            else:
+                boolValid = True
+        new_donor.append({"donor name": donor_full_name, "total donations": new_donation_amount, "number donations": 1,
+                          "avg donation": new_donation_amount})
+        lst_Donors.extend(new_donor)
+        print('-----------------------------------------------------')
+        for x in lst_Donors:
+            if x["donor name"] == donor_full_name:
+                email = "Dear {a},\n\nThank you for your generous donations of ${b:.2f} to our charity.\n".format(
+                    a=x["donor name"],b=x["total donations"])
+                print(email)
+    print('-----------------------------------------------------')
+
+
+def sort_list(lst_Donors):
+    # used to sort the donor list by the total donations column
+
+    return lst_Donors["total donations"]
+
+
+def create_report():
+    # creates a report of the the donors
+    # headers used in table
+    print()
+    lst_Header = [["Donor Name", "| Total Donation(s)", "| # of Donations", "| Avg Donation"]]
+
+    for x in lst_Header:
+        print('{:<25}{:<20}{:<17}{:<15}'.format(*x))
+    print("----------------------------------------------------------------------------")
+    for x in sorted(lst_Donors,key=sort_list, reverse= True):
+        print('{:<25} $ {:<20}{:^14} $ {:<15}'.format(*x.values()))
+
+
+def send_letter_all():
+    print('\n----------------------------------------------')
+    file = ()
```

Comment:
don't use `file` because you will overwrite built-in `file`. github highlights built-in keywords.